pkg/arguments/arguments.go

```diff
@@ -185,6 +187,91 @@ func ParseKnownFlags(cmd *cobra.Command, argv []string, failOnUnknown bool) erro
 	return nil
 }
 
+// PreprocessUnknownFlagsWithId Parses known and unknown flags will take the command line arguments and map the ones
+// that fit with known flags.
+func PreprocessUnknownFlagsWithId(cmd *cobra.Command, argv []string) error {
+	flags := cmd.Flags()
+
+	var validArgs []string
+	var upcomingValue bool
+
+	foundId := false
+	for _, arg := range argv {
+		switch {
+		// Upcoming value from a space-separated value
+		case upcomingValue:
+			if strings.HasPrefix(arg, "-") {
+				return fmt.Errorf("No value given for flag '%s'", arg)
```

Comment:
@hunterkepley Does this error need to refer to the previous value of `arg`? If I follow correctly, you expect `arg` to be the value of the previous flag, but here we determine that `arg` is another flag. So there is no value given for the previous `arg` and not the current `arg`.  So as an example:

```
./rosa edit addon -foo= -bar=baz
```

Where we expect `-foo` and `-bar` to have values. I _think_ the error statement here will log:

`No value given for the flag '-bar'` as you're relying on the current value of `arg`.

Reply:
This is the case, but i ended up removing this check as it is not needed, cobra will give a proper error anyway without this :+1: